infra/util/src/main/java/org/apache/shardingsphere/infra/util/yaml/constructor/ShardingSphereYamlConstructor.java

```diff
@@ -49,6 +49,7 @@ public ShardingSphereYamlConstructor(final Class<?> rootClass) {
     
     private static LoaderOptions createLoaderOptions() {
         LoaderOptions result = new LoaderOptions();
+        result.setMaxAliasesForCollections(Integer.MAX_VALUE);
```

Comment:
And could we set `maxAliasesForCollections` value to limited value? e.g. `1000`. Since
1) It's enough for now.
2) Use `Integer.MAX_VALUE` might have vulnerability, see:
- https://github.com/spring-projects/spring-boot/issues/23096
- https://en.wikipedia.org/wiki/Billion_laughs_attack

Reply:
And could we disable anchor and alias in snakeyaml?

Reply:
Refer
> https://stackoverflow.com/questions/63823021/how-to-auto-edit-yaml-file-containing-anchors-aliases-using-snakeyaml
https://www.spigotmc.org/threads/yaml-anchors-and-aliases-how-to-disable-them.523792/

At latest snakeyaml version, should be no option provided to disable aliases directly

Reply:
> And could we set `maxAliasesForCollections` value to limited value? e.g. `1000`. Since
> 
> 1. It's enough for now.
> 2. Use `Integer.MAX_VALUE` might have vulnerability, see:
> 
> * [Large files can throw YAMLException due to node limits spring-projects/spring-boot#23096](https://github.com/spring-projects/spring-boot/issues/23096)
> * https://en.wikipedia.org/wiki/Billion_laughs_attack

And LoaderOptions.allowRecursiveKeys maybe could avoid the problem
<img width="906" alt="image" src="https://github.com/apache/shardingsphere/assets/101622833/e53d40d1-cea4-455c-b97d-869ac74e793e">

Reply:
OK